github/github.go

```diff
@@ -174,6 +174,10 @@ type Client struct {
 	rateLimits              [Categories]Rate // Rate limits for the client as determined by the most recent API calls.
 	secondaryRateLimitReset time.Time        // Secondary rate limit reset for the client as determined by the most recent API calls.
 
+	// If specified, Client will block requests for at most this duration in case of reaching a secondary
+	// rate limit
+	MaxSecondaryRateLimitRetryAfterDuration time.Duration
```

Comment:
I'm letting users specify their desired max duration. We could enforce a minimum or a default value here if we want to make sure users are not using this field incorrectly